tools/static-data/generate-lambda-node-schedule.mjs

```diff
@@ -0,0 +1,53 @@
+import { z } from 'zod';
+import { updateJsonFile } from './utils.mjs';
+
+const RuntimesSchema = z.array(
+  z.object({
+    cycle: z.string().describe('The ID of the Runtime'),
+    support: z
+      .union([z.boolean(), z.string()])
+      .describe(
+        'Either `true` if in support, or a string denoting when support for this Runtime will end. 0.10.x is a sole exception which has a value of `false` and will be filtered out',
+      ),
+  }),
+);
+
+const lambdaDataUrl = 'https://endoflife.date/api/aws-lambda.json';
+
+await (async () => {
+  console.log('Generating node schedule');
+
+  const lambdas = await fetch(lambdaDataUrl).then(async (response) => {
+    if (!response.ok) {
+      console.error(`Failed to fetch ${lambdaDataUrl}`, response);
+      process.exit(1);
+    }
+
+    return RuntimesSchema.parseAsync(await response.json());
+  });
+
+  const nodeRuntimes = z.record(z.string(), RuntimesSchema.element).parse({});
```

Comment:
I had to do a `.parse()` here because typescript is still inferring a type, even in JS, and `nodeRuntimes[versionMatch[1]] = lambda;` wanted `nodeRuntimes` to be better typed. Before I was using the JSDocs parameters, but moving to zod I couldn't find a better public interface for getting the output type. This lets typescript infer the correct typing of this record

Reply:
what about this?
```suggestion
  /** @type {Record<string,RuntimesSchema>} */
  const nodeRuntimes = {};
```

Reply:
That was the first thing I tried and it doesn't work. `nodeRuntimes[versionMatch[1]] = lambda;` ends up complaining because it expects `lambda` to match the `ZodObject` typing, which contains extra properties. I couldn't figure out a way to get it to type-hint against the original shape that zod is wrapping instead of the `ZodObject` type. Here's the type error when I try it:

```
error TS2740: Type '{ support: string | boolean; cycle: string; }' is missing the following properties from type 'ZodObject<{ cycle: ZodString; support: ZodUnion<[ZodBoolean, ZodString]>; }, "strip", ZodTypeAny, { support: string | boolean; cycle: string; }, { ...; }>': _cached, _getCached, _parse, shape, and 52 more.
```

Reply:
and what about this? (needs other change above)
```suggestion
  /** @type {Record<string, z.infer<RuntimesSchema>>} */
  const nodeRuntimes = {};
```